google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkCommitBatch.java

```diff
@@ -69,40 +69,37 @@ ApiFuture<WriteResult> wrapResult(int writeIndex) {
    * <p>The writes in the batch are not applied atomically and can be applied out of order.
    */
   ApiFuture<Void> bulkCommit() {
+
+    // Follows same thread safety logic as `UpdateBuilder::commit`.
+    committed = true;
+    BatchWriteRequest request = buildBatchWriteRequest();
+
     Tracing.getTracer()
         .getCurrentSpan()
         .addAnnotation(
             TraceUtil.SPAN_NAME_BATCHWRITE,
-            ImmutableMap.of("numDocuments", AttributeValue.longAttributeValue(getWrites().size())));
-
-    final BatchWriteRequest.Builder request = BatchWriteRequest.newBuilder();
-    request.setDatabase(firestore.getDatabaseName());
-
-    for (WriteOperation writeOperation : getWrites()) {
-      request.addWrites(writeOperation.write);
-    }
-
-    committed = true;
+            ImmutableMap.of(
+                "numDocuments", AttributeValue.longAttributeValue(request.getWritesCount())));
 
     ApiFuture<BatchWriteResponse> response =
         processExceptions(
-            firestore.sendRequest(request.build(), firestore.getClient().batchWriteCallable()));
+            firestore.sendRequest(request, firestore.getClient().batchWriteCallable()));
 
     return ApiFutures.transformAsync(
         response,
         batchWriteResponse -> {
-          List<ApiFuture<Void>> pendingUserCallbacks = new ArrayList<>();
-
           List<com.google.firestore.v1.WriteResult> writeResults =
               batchWriteResponse.getWriteResultsList();
           List<com.google.rpc.Status> statuses = batchWriteResponse.getStatusList();
 
-          for (int i = 0; i < writeResults.size(); ++i) {
-            com.google.firestore.v1.WriteResult writeResult = writeResults.get(i);
+          int size = writeResults.size();
```

Comment:
It doesn't look like the changes in `transformAsync()` are fixing any concurrency bugs. Can these changes be reverted in this PR? Fully addressing the concurrency bugs in `BulkCommitBatch` is for another PR IMO.

Reply:
I'll revert.